install/infra/single-cluster/gcp/README.md

```diff
@@ -0,0 +1,235 @@
+```
+
+The `apply` target calls the `terraform` apply on `eks` module, `cert-manager`
+module, `external-dns` module and `cluster-issuer` module in that exact order.
+The entire operation would take around *30 minutes* to complete.
+
+Upon completion, you will find a creation `kubeconfig` file in the local
+directory. You can try accessing the cluster using this file as follows:
+
+``` sh
+export KUBECONFIG=/path/to/kubeconfig
+kubectl get pods -A
+```
+
+You can list all the other outputs with the following command:
+
+``` sh
+make output
+```
+
+> 💡 Alternatively, you can get the simple JSON output with a `terraform output` command
+
+## Note the NS records from terraform output
+
+Once the apply process has exited successfully, we can go ahead and prepare to
+setup Gitpod. If you specified the `domain_name` in the `terraform.tfvars` file,
+the terraform module registers the module with `cloudDNS` to point to the
+cluster. Now you have to configure whichever provider you use to host your
+domain name to route traffic to the GCP name servers. You can find these name
+servers in the `make output` command from above. It would be of the format:
+
+```json
+Nameservers for the domain(to be added as NS records in your domain provider):
+=================
+[
+  "ns-cloud-c1.googledomains.com.",
+  "ns-cloud-c2.googledomains.com.",
+  "ns-cloud-c3.googledomains.com.",
+  "ns-cloud-c4.googledomains.com."
+]
+
+
+```
+
+Add the `ns` records similar to the above 4 URIs as NS records under your domain
+DNS management setup. Check with your domain hosting service for specific information.
+
+## Note the dependency credentials from terraform output
+
+If you enabled the creation of external database, storage and registry, the
+above `make output` command would list the credentials to connect to the
+corresponding resource. Make a note of this, so as to provide the same when
+setting up Gitpod via KOTS UI.
+
+## Install Gitpod
+
+You can install `KOTS` CLI to install Gitpod:
+
+``` sh
+curl https://kots.io/install | bash
+```
+
+Run the following to get started with Gitpod installation:
+
+``` sh
+export KUBECONFIG=kubeconfig
+kubectl kots install gitpod/stable # you can replace `stable` with `unstable` or `beta` as per the requirement
+```
+
+Upon completion, you can access `KOTS` UI in `localhost:8800`. Here you can
+proceed to configuring and intalling Gitpod. Please follow the [official
+documentaion](https://www.gitpod.io/docs/self-hosted/latest/getting-started#step-4-install-gitpod) to complete the Gitpod setup.
+
+## Troubleshooting
+
+### Some pods never start (Init state)
+
+```sh
+kubectl get pods -l component=proxy
+NAME                     READY   STATUS    RESTARTS   AGE
+proxy-5998488f4c-t8vkh   0/1     Init 0/1  0          5m
+```
+
+The most likely reason is that the DNS01 challenge has yet to resolve. To fix this, make sure you have added the NS records corresponding to the `cloudDNS` zone of the `domain_name` added to your domain provider.
```

Comment:
**Opinion:** To me, "has yet to resolve" means that this issue should be able to resolve by itself. WDYT of the following?

```suggestion
The most likely reason is that the DNS01 challenge cannot be completed, typically because DNS zone delegation hasn't been set up from the parent domain to the subdomain that Gitpod is managing (specified by the `domain_name` variable). To fix this, make sure that NS records for `domain_name` in the parent zone are created and point to the nameservers of the Gitpod managed zone. See the [Google Cloud DNS documentation](https://cloud.google.com/dns/docs/dns-overview#delegated_subzone) for more information on zone delegation.
```